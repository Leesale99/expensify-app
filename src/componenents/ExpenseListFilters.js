import React from 'react';
import { connect } from 'react-redux';
import {
  setTextFilter,
  sortByAmount,
  sortByDate,
  setStartDate,
  setEndDate
} from '../actions/filters';
import { DateRangePicker } from 'react-dates';

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  };

  onTextChange = e => this.props.setTextFilter(e.target.value);

  handleSortBy = e => {
    switch (e.target.value) {
      case 'date':
        this.props.sortByDate();
        break;
      case 'amount':
        this.props.sortByAmount();
      default:
        break;
    }
  };

  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };

  onFocusChange = calendarFocused => {
    this.setState(() => ({ calendarFocused }));
  };

  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={this.onTextChange}
        />
        <select value={this.props.filters.sortBy} onChange={this.handleSortBy}>
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => false}
          showClearDates={true}
        />
      </div>
    );
  }
}

const mapToStateProps = state => ({
  filters: state.filters
});

const mapDispatchToProps = dispatch => ({
  setTextFilter: text => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setStartDate: startDate => dispatch(setStartDate(startDate)),
  setEndDate: endDate => dispatch(setEndDate(endDate))
});

export default connect(mapToStateProps, mapDispatchToProps)(ExpenseListFilters);
