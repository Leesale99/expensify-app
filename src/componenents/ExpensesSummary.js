import React from 'react';
import { connect } from 'react-redux';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = props => {
  const info = `Viewing ${props.expenseCount} ${props.expenseCount === 1
    ? 'expense'
    : 'expenses'} totaling ${numeral(props.expensesTotal / 100).format(
    '$0,0.00'
  )}`;
  return (
    <div>
      <h2>Total:</h2>
      <p>{info}</p>
    </div>
  );
};

const mapStateToProps = state => ({
  expenseCount: state.expenses.length,
  expensesTotal: selectExpensesTotal(state.expenses)
});

export default connect(mapStateToProps)(ExpensesSummary);
