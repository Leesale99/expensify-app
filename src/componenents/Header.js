import React from 'react';
import { NavLink } from 'react-router-dom';
import { startLogout } from '../actions/auth';
import { connect } from 'react-redux';

export const Header = ({ startLogout }) => (
  <header>
    <h1>Expensify</h1>
    <ul>
      <li>
        <NavLink to="/" activeClassName="is-active" exact={true}>
          Login
        </NavLink>
      </li>
      <li>
        <NavLink to="/dashboard" activeClassName="is-active">
          Dashboard
        </NavLink>
      </li>
      <li>
        <NavLink to="/create" activeClassName="is-active">
          Add Expense
        </NavLink>
      </li>
      <li>
        <button onClick={startLogout}>Logout</button>
      </li>
    </ul>
  </header>
);

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
