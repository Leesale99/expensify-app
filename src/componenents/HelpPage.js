import React from 'react';

const HelpPage = () => <div>this is Help</div>;

export default HelpPage;
