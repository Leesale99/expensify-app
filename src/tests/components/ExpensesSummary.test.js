import React from 'react';
import { ExpensesSummary } from '../../componenents/ExpensesSummary';
import { shallow } from 'enzyme';
import selectExpensesTotal from '../../selectors/expenses-total';
import expenses from '../fixtures/expenses';

test('should render ExpenseSummary with 0 expenses correctly', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={0} expensesTotal={0} />
  );

  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary with multiple expenses correctly', () => {
  const expenseCount = expenses.length;
  const expensesTotal = selectExpensesTotal(expenses);
  const wrapper = shallow(
    <ExpensesSummary
      expenseCount={expenseCount}
      expensesTotal={expensesTotal}
    />
  );

  expect(wrapper).toMatchSnapshot();
});
