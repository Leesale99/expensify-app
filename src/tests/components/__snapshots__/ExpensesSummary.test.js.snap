// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with 0 expenses correctly 1`] = `
<div>
  <h1>
    Viewing 0 expenses totaling $0.00
  </h1>
</div>
`;

exports[`should render ExpenseSummary with multiple expenses correctly 1`] = `
<div>
  <h1>
    Viewing 3 expenses totaling $1,141.95
  </h1>
</div>
`;
